pkg/scheduler/scheduler.go

```diff
@@ -522,8 +522,8 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 	assumedPodInfo := podInfo.DeepCopy()
 	assumedPod := assumedPodInfo.Pod
 
-	// Run "reserve" plugins.
-	if sts := prof.RunReservePlugins(schedulingCycleCtx, state, assumedPod, scheduleResult.SuggestedHost); !sts.IsSuccess() {
+	// Run the Reserve method of reservation plugins.
+	if sts := prof.RunReserveReservationPlugins(schedulingCycleCtx, state, assumedPod, scheduleResult.SuggestedHost); !sts.IsSuccess() {
 		sched.recordSchedulingFailure(prof, assumedPodInfo, sts.AsError(), SchedulerError, "")
```

Comment:
Here we should call Unreserve #83557

We will need a test for it.

Reply:
I'd like to implement the merge first and then handle the case of failing reserves later separately to keep the PR small, atomic, and contained to one issue.

Reply:
Could I recreate https://github.com/kubernetes/kubernetes/pull/83557 after this PR will be merged?

Reply:
@adtac could take it, since he is already modifying the code here. Unless you @mrkm4ntr  have some specific need?

Reply:
I'd like to see this feature to completion, but if you'd strongly like to work on that feature after this PR, I don't mind :)

Reply:
Oh, I misunderstood. I have nothing to do :)